posts/2023-05-24-api-automation-testing-tools-comparison.adoc

```diff
@@ -0,0 +1,320 @@
+TehreemFatima07
+v1.0, 2023-05-24
+:title: API Automation Testing Tools, a comparison
+:imagesdir: ../media/2023-05-24-api-automation-testing-tools-comparison
+:lang: en
+:tags: [api, test automation, testing techniques]
+
+
+Internal projects at Lunatech play a crucial role in fostering growth, innovation, and knowledge-sharing within employees. These projects serve as the sandbox where new joiners are given the opportunity to explore, learn, and contribute in a hands-on manner.
+++++
+</br>
+++++
+Lunagraph is one of the internal projects that will revolutionize how we handle three crucial elements: People, Clients, and Projects. This innovative solution serves as the "source of truth" for these key elements, providing a reliable and consolidated platform that offers immense value to our internal clients.
+++++
+</br>
+++++
+Lunagraph's vision is to be able to reliably service internal clients with a single API integration, previously many apps were used for this integration. One of the task for this project is to have automated testing of APIs for faster test maintenance and refactoring.
+
+=== What is an API Testing?
+API testing refers to the practice of testing Application Programming Interfaces (APIs) to ensure their functionality, reliability, security, and performance. It allows software teams to improve APIs without the worry of introducing bugs or disrupting current functionalities. The process of upgrading code dependencies becomes easier when API verification can be performed automatically.
+
+To start with the API Testing, first step was to select the API Testing tool. This blog post explores the API Testing tools, to present the strengths and weakness of them and to give the comparison that which tools best fits the purpose of API Testing of Lunagraph
+
+=== 1. Postman
+Postman is a collaboration platform for API development. It is a popular API client and it allows us to design, build, share, test, and document APIs. Key features of the Postman include:
+
+- Easy-to-use REST client
+- Available for both automated and exploratory testing
+- Provide many integrations like support for Swagger & RAML formats
+- Run, test, document, and monitoring features
+- Black-box style test - Postman (and Newman) do not require direct access to the code for the System under Test (SUT), as they test it from the outside. It's ideally suited to exploring and collaborating on APIs and running functional API tests.
+
+Sample code for writing a test in Postman:
+[source,javascript]
+----
+// Perform assertions on the response
+pm.test('Status code is 200', function () {
+  pm.response.to.have.status(200);
+});
+
+pm.test('Response has valid data', function () {
+  var jsonData = pm.response.json();
+  pm.expect(jsonData).to.have.property('key', 'value');
+});
+----
+
+`Newman – Newman is Postman’s interface that runs a battery of tests from the command line even if the GUI (Graphical User Interface) is not ready`
```

Comment:
While this is not wrong, I feel that better description of Newman could be written here. This following one i from Postman official website (https://learning.postman.com/docs/collections/using-newman-cli/command-line-integration-with-newman/):
`Newman is a command-line Collection Runner for Postman. It enables you to run and test a Postman Collection directly from the command line. It's built with extensibility in mind so that you can integrate it with your continuous integration (CI) servers and build systems.`

Maybe worth considering.

Reply:
Yes, I have added this to the definition as well